test/libsolidity/syntaxTests/blobhash_shadow_warning.sol

```diff
@@ -0,0 +1,12 @@
+contract C {
+    function blobhash(uint256 index) public pure returns(bytes32) {
+        return bytes32(index);
+    }
+    function f() public pure returns(bytes32) {
+        return blobhash(3);
+    }
+}
+// ====
+// EVMVersion: >=cancun
```

Comment:
A copy of this test with ``<cancun`` would also be nice

Reply:
Indeed, looking at it again I realized a mistake. Currently, `blobhash` would be exposed in the global context, even for EVM versions `<cancun`, as it is unrestrictedly added as a `MagicVariable`, consequently being considered a built-in. Therefore, we may need to pass the `evmVersion` to `GlobalContext` or add exceptions whenever it is used.

Reply:
Or if this is the expected behavior, we should consider removing the `EVMVersion` settings from this test and let it throw the same warning for all EVM versions.

Reply:
How much effort is it to add `EVMVersion` there? It's probably what we'll need to do in the end.

Reply:
If doing that is non-trivial, I think that for this PR it would be good enough to ensure that trying to call the global `blockhash()` on `<cancun` produces an error. We could leave dealing with the spurious shadowing warning for a follow-up PR, since that's mostly cosmetic.

Reply:
Yeah, not much. I was just trying to see if there was a better alternative, because there are some helper functions that I will need to change as well, like [constructMagicVariables()](https://github.com/ethereum/solidity/blob/1473ce21f37bffb57889b8729959acc02888ccc9/libsolidity/analysis/GlobalContext.cpp#L66C69-L66C92) and [magicVariableToID()](https://github.com/ethereum/solidity/blob/1473ce21f37bffb57889b8729959acc02888ccc9/libsolidity/analysis/GlobalContext.cpp#L38). So I will just add `evmVersion` as member of `GlobalContext` and make those functions private methods of it.

Reply:
Just make sure that there's no possibility `GlobalContext` will be reused for a different EVM version. It's global after all :)